lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -882,158 +878,171 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
             Cardano.TxReturnCollateral _ _ ->
                throwE ErrBalanceTxExistingReturnCollateral
 
-    -- | Select assets to cover the specified balance and fee.
-    --
-    -- If the transaction contains redeemers, the function will also ensure the
-    -- selection covers the fees for the maximum allowed execution units of a
-    -- transaction. For this, and other reasons, the selection may include too
-    -- much ada.
-    selectAssets'
-        :: Cardano.AnyCardanoEra
-        -> [W.TxOut]
-        -> UTxOSelection WalletUTxO
-        -- ^ Describes which utxos are pre-selected, and which can be used as
-        -- inputs or collateral.
-        -> Cardano.Value -- Balance to cover
-        -> Cardano.Lovelace -- Current minfee (before selecting assets)
-        -> StdGenSeed
-        -> Either (SelectionError WalletSelectionContext) Selection
-    selectAssets' era outs utxoSelection balance fee0 seed =
-        flip evalRand (stdGenFromSeed seed)
-            $ runExceptT
+    fromCardanoLovelace (Cardano.Lovelace l) = Coin.unsafeFromIntegral l
+
+-- | Select assets to cover the specified balance and fee.
+--
+-- If the transaction contains redeemers, the function will also ensure the
+-- selection covers the fees for the maximum allowed execution units of a
+-- transaction. For this, and other reasons, the selection may include too
+-- much ada.
+selectAssets
+    :: forall era changeState
+     . Cardano.IsCardanoEra era
+    => RecentEra era
+    -> ProtocolParameters era
+    -> TxWitnessTag
```

Comment:
IMO we might as well take `UTxOAssumptions`. The reason is that of the four `txLayer toInpScriptsM mScriptTemplate txWitnessTag` fields, three are used by `selectAssets` and only `toInpScriptsM` is used by `balanceTransaction`, where I also suspect it would make sense if the input script lookup also were moved to  `selectAssets`. I.e. I think `UTxOAssumptions` is more related to `selectAssets` than to `balanceTransaction`. 

 (I'd fine with merging as is though, as this function still only is called by balanceTx.)